### PR TITLE
Update Assn5

### PR DESCRIPTION
**Pull request by TlucasX5**
The html page is not returning the diamond pattern - gives me a "undefined error". Not sure what I'm missing...?
Also I feel like this assignment - in comparison to the others so far - went from 0 to 100 in complexity. Took me a long time to figure out compared to a basic triangle for example. I still have to take college algebra so maybe that's why I'm struggling with the math and need to research formulas. I am reading the book (did we skip chapters? I can't tell where we are based on the notes in Brightspace) but is there another book you would recommend for learning?

### CODE REVIEW

**Comment by akennis** [comment]
Thanks for your effort and sorry for the trouble.  I'm reviewing your code now.  Let's discuss before or in class today please.

File: assignments/Assignment5.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// Use nested loops to create diamond pattern
+
+function diamondPattern(centerWidth) {
```
Comment (akennis) [inline assignments/Assignment5.js:3]: This function works just fine - I can see it generates the diamonds of the correct width.  It should though `return s;` as its final statement.  The caller can then receive the diamond patter back and either log it or put it into the HTML doc for display there.

File: assignments/Assignment5.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    //let centerWidth = 5;
+    let s = "";
+
+    // Build top of diamond
+    for (i=1; i<=centerWidth; i+=2) {
```
Comment (akennis) [inline assignments/Assignment5.js:8]: nit: variables declared within for loops should be declared using the `let` keyword

File: assignments/Assignment5.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      
+      for (j=centerWidth; j>i; j-=2) {
```
Comment (akennis) [inline assignments/Assignment5.js:10]: Here too

File: assignments/Assignment5.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+          s += " ";
+      }
+
+      for (k=0; k<i*2-1; k+=2) {
```
Comment (akennis) [inline assignments/Assignment5.js:14]: And here too

File: assignments/Assignment5.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+      for (k=(centerWidth-i)*2-1; k>1; k-=2) {
+          s += "*";
+      }
+
+      s+= "\n";
+
+    }
+
+    console.log(s);
```
Comment (akennis) [inline assignments/Assignment5.js:38]: Remove this and instead `return s;`

File: assignments/Assignment5.html [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <script src="Assignment5.js"></script>
+</head>
+<body>
+    <input id="centerWidth" type="text" placeholder="centerWidth"></input><br>
+    <button onclick="diamondPatternOnClickHandler()">Get diamond pattern</button><br>
+    <input id="res" type="text"></input>
```
Comment (akennis) [inline assignments/Assignment5.html:9]: This should be a `<textarea...>` instead:

https://www.w3schools.com/tags/tag_textarea.asp

**Comment by akennis** [comment]
> Thanks for your effort and sorry for the trouble. I'm reviewing your code now. Let's discuss before or in class today please.

I reviewed your code now - it's very good actually, certainly better than you described yourself!  I left a few comments for minor things to fixup.  As it stands I have this down for a 90%.

**Comment by TlucasX5** [comment]
will add changes

**Pull request closed without merging**